
From: Rik van Riel <riel@redhat.com>

Here you are.  The following patch replaces the numbers with
a define called MLOCK_LIMIT.

Signed-off-by: Rik van Riel <riel@redhat.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/include/asm-alpha/resource.h   |    2 +-
 25-akpm/include/asm-arm/resource.h     |    2 +-
 25-akpm/include/asm-arm26/resource.h   |    2 +-
 25-akpm/include/asm-cris/resource.h    |    2 +-
 25-akpm/include/asm-h8300/resource.h   |    2 +-
 25-akpm/include/asm-i386/resource.h    |    2 +-
 25-akpm/include/asm-ia64/resource.h    |    2 +-
 25-akpm/include/asm-m68k/resource.h    |    2 +-
 25-akpm/include/asm-mips/resource.h    |    2 +-
 25-akpm/include/asm-parisc/resource.h  |    2 +-
 25-akpm/include/asm-ppc/resource.h     |    2 +-
 25-akpm/include/asm-ppc64/resource.h   |    2 +-
 25-akpm/include/asm-s390/resource.h    |    2 +-
 25-akpm/include/asm-sh/resource.h      |    2 +-
 25-akpm/include/asm-sparc/resource.h   |    2 +-
 25-akpm/include/asm-sparc64/resource.h |    2 +-
 25-akpm/include/asm-v850/resource.h    |    2 +-
 25-akpm/include/asm-x86_64/resource.h  |    2 +-
 25-akpm/include/linux/resource.h       |    6 ++++++
 19 files changed, 24 insertions(+), 18 deletions(-)

diff -puN include/asm-alpha/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-alpha/resource.h
--- 25/include/asm-alpha/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.747964920 -0700
+++ 25-akpm/include/asm-alpha/resource.h	2004-08-05 11:16:01.777960360 -0700
@@ -41,7 +41,7 @@
     {INR_OPEN, INR_OPEN},			/* RLIMIT_NOFILE */	\
     {LONG_MAX, LONG_MAX},			/* RLIMIT_AS */		\
     {LONG_MAX, LONG_MAX},			/* RLIMIT_NPROC */	\
-    {32768, 	32768	},			/* RLIMIT_MEMLOCK */	\
+    {MLOCK_LIMIT, MLOCK_LIMIT },		/* RLIMIT_MEMLOCK */	\
     {LONG_MAX, LONG_MAX},			/* RLIMIT_LOCKS */	\
     {MAX_SIGPENDING, MAX_SIGPENDING},		/* RLIMIT_SIGPENDING */ \
     {MQ_BYTES_MAX, MQ_BYTES_MAX},		/* RLIMIT_MSGQUEUE */	\
diff -puN include/asm-arm26/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-arm26/resource.h
--- 25/include/asm-arm26/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.748964768 -0700
+++ 25-akpm/include/asm-arm26/resource.h	2004-08-05 11:16:01.779960056 -0700
@@ -39,7 +39,7 @@
 	{ RLIM_INFINITY, RLIM_INFINITY },	\
 	{ 0,             0             },	\
 	{ INR_OPEN,      INR_OPEN      },	\
-	{ 32768,	 32768	       },	\
+	{ MLOCK_LIMIT,	 MLOCK_LIMIT   },	\
 	{ RLIM_INFINITY, RLIM_INFINITY },	\
 	{ RLIM_INFINITY, RLIM_INFINITY },	\
 	{ MAX_SIGPENDING, MAX_SIGPENDING},	\
diff -puN include/asm-arm/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-arm/resource.h
--- 25/include/asm-arm/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.749964616 -0700
+++ 25-akpm/include/asm-arm/resource.h	2004-08-05 11:16:01.776960512 -0700
@@ -39,7 +39,7 @@
 	{ RLIM_INFINITY, RLIM_INFINITY },	\
 	{ 0,             0             },	\
 	{ INR_OPEN,      INR_OPEN      },	\
-	{ 32768,	 32768	       },	\
+	{ MLOCK_LIMIT,	 MLOCK_LIMIT   },	\
 	{ RLIM_INFINITY, RLIM_INFINITY },	\
 	{ RLIM_INFINITY, RLIM_INFINITY },	\
 	{ MAX_SIGPENDING, MAX_SIGPENDING},	\
diff -puN include/asm-cris/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-cris/resource.h
--- 25/include/asm-cris/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.751964312 -0700
+++ 25-akpm/include/asm-cris/resource.h	2004-08-05 11:16:01.778960208 -0700
@@ -39,7 +39,7 @@
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{             0,             0 },		\
 	{      INR_OPEN,     INR_OPEN  },		\
-	{         32768,         32768 },               \
+	{   MLOCK_LIMIT,   MLOCK_LIMIT },               \
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ MAX_SIGPENDING, MAX_SIGPENDING },		\
diff -puN include/asm-h8300/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-h8300/resource.h
--- 25/include/asm-h8300/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.752964160 -0700
+++ 25-akpm/include/asm-h8300/resource.h	2004-08-05 11:16:01.778960208 -0700
@@ -39,7 +39,7 @@
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{             0,             0 },		\
 	{      INR_OPEN,     INR_OPEN  },		\
-	{         32768,         32768 },		\
+	{   MLOCK_LIMIT,   MLOCK_LIMIT },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ MAX_SIGPENDING, MAX_SIGPENDING },		\
diff -puN include/asm-i386/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-i386/resource.h
--- 25/include/asm-i386/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.753964008 -0700
+++ 25-akpm/include/asm-i386/resource.h	2004-08-05 11:16:01.777960360 -0700
@@ -40,7 +40,7 @@
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{             0,             0 },		\
 	{      INR_OPEN,     INR_OPEN  },		\
-	{         32768,         32768 },		\
+	{   MLOCK_LIMIT,   MLOCK_LIMIT },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ MAX_SIGPENDING, MAX_SIGPENDING },		\
diff -puN include/asm-ia64/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-ia64/resource.h
--- 25/include/asm-ia64/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.755963704 -0700
+++ 25-akpm/include/asm-ia64/resource.h	2004-08-05 11:16:01.776960512 -0700
@@ -46,7 +46,7 @@
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{             0,             0 },		\
 	{      INR_OPEN,     INR_OPEN  },		\
-	{         32768,         32768 },		\
+	{   MLOCK_LIMIT,   MLOCK_LIMIT },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ MAX_SIGPENDING, MAX_SIGPENDING },		\
diff -puN include/asm-m68k/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-m68k/resource.h
--- 25/include/asm-m68k/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.756963552 -0700
+++ 25-akpm/include/asm-m68k/resource.h	2004-08-05 11:16:01.775960664 -0700
@@ -39,7 +39,7 @@
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{             0,             0 },		\
 	{      INR_OPEN,     INR_OPEN  },		\
-	{         32768,         32768 },		\
+	{   MLOCK_LIMIT,   MLOCK_LIMIT },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ MAX_SIGPENDING, MAX_SIGPENDING },		\
diff -puN include/asm-mips/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-mips/resource.h
--- 25/include/asm-mips/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.757963400 -0700
+++ 25-akpm/include/asm-mips/resource.h	2004-08-05 11:16:01.774960816 -0700
@@ -53,7 +53,7 @@
 	{ INR_OPEN,      INR_OPEN      },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
-	{ 0,             0             },		\
+	{ MLOCK_LIMIT,     MLOCK_LIMIT },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ MAX_SIGPENDING, MAX_SIGPENDING },		\
diff -puN include/asm-parisc/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-parisc/resource.h
--- 25/include/asm-parisc/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.759963096 -0700
+++ 25-akpm/include/asm-parisc/resource.h	2004-08-05 11:16:01.775960664 -0700
@@ -39,7 +39,7 @@
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{             0,             0 },		\
 	{      INR_OPEN,     INR_OPEN  },		\
-	{         32768,         32768 },		\
+	{   MLOCK_LIMIT,   MLOCK_LIMIT },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ MAX_SIGPENDING, MAX_SIGPENDING },		\
diff -puN include/asm-ppc64/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-ppc64/resource.h
--- 25/include/asm-ppc64/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.760962944 -0700
+++ 25-akpm/include/asm-ppc64/resource.h	2004-08-05 11:16:01.774960816 -0700
@@ -45,7 +45,7 @@
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{             0,             0 },		\
 	{      INR_OPEN,     INR_OPEN  },		\
-	{         32768,         32768 },		\
+	{   MLOCK_LIMIT,   MLOCK_LIMIT },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ MAX_SIGPENDING, MAX_SIGPENDING },		\
diff -puN include/asm-ppc/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-ppc/resource.h
--- 25/include/asm-ppc/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.761962792 -0700
+++ 25-akpm/include/asm-ppc/resource.h	2004-08-05 11:16:01.775960664 -0700
@@ -36,7 +36,7 @@
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{             0,             0 },		\
 	{      INR_OPEN,     INR_OPEN  },		\
-	{         32768,         32768 },		\
+	{   MLOCK_LIMIT,   MLOCK_LIMIT },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ MAX_SIGPENDING, MAX_SIGPENDING },		\
diff -puN include/asm-s390/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-s390/resource.h
--- 25/include/asm-s390/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.763962488 -0700
+++ 25-akpm/include/asm-s390/resource.h	2004-08-05 11:16:01.778960208 -0700
@@ -47,7 +47,7 @@
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{             0,             0 },		\
 	{ INR_OPEN, INR_OPEN },                         \
-	{         32768,         32768 },		\
+	{   MLOCK_LIMIT,   MLOCK_LIMIT },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ MAX_SIGPENDING, MAX_SIGPENDING },		\
diff -puN include/asm-sh/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-sh/resource.h
--- 25/include/asm-sh/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.764962336 -0700
+++ 25-akpm/include/asm-sh/resource.h	2004-08-05 11:16:01.779960056 -0700
@@ -39,7 +39,7 @@
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{             0,             0 },		\
 	{      INR_OPEN,     INR_OPEN  },		\
-	{         32768,         32768 },		\
+	{   MLOCK_LIMIT,   MLOCK_LIMIT },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ MAX_SIGPENDING, MAX_SIGPENDING },		\
diff -puN include/asm-sparc64/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-sparc64/resource.h
--- 25/include/asm-sparc64/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.765962184 -0700
+++ 25-akpm/include/asm-sparc64/resource.h	2004-08-05 11:16:01.779960056 -0700
@@ -43,7 +43,7 @@
     {       0, RLIM_INFINITY},		\
     {RLIM_INFINITY, RLIM_INFINITY},	\
     {INR_OPEN, INR_OPEN}, {0, 0},	\
-    {32768, 	  32768	          },	\
+    {  MLOCK_LIMIT,   MLOCK_LIMIT},	\
     {RLIM_INFINITY, RLIM_INFINITY},	\
     {RLIM_INFINITY, RLIM_INFINITY},	\
     {MAX_SIGPENDING, MAX_SIGPENDING},	\
diff -puN include/asm-sparc/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-sparc/resource.h
--- 25/include/asm-sparc/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.767961880 -0700
+++ 25-akpm/include/asm-sparc/resource.h	2004-08-05 11:16:01.777960360 -0700
@@ -44,7 +44,7 @@
     {       0, RLIM_INFINITY},		\
     {RLIM_INFINITY, RLIM_INFINITY},	\
     {INR_OPEN, INR_OPEN}, {0, 0},	\
-    {32768, 	     32768},	\
+    {MLOCK_LIMIT,   MLOCK_LIMIT},	\
     {RLIM_INFINITY, RLIM_INFINITY},	\
     {RLIM_INFINITY, RLIM_INFINITY},	\
     {MAX_SIGPENDING, MAX_SIGPENDING},	\
diff -puN include/asm-v850/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-v850/resource.h
--- 25/include/asm-v850/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.768961728 -0700
+++ 25-akpm/include/asm-v850/resource.h	2004-08-05 11:16:01.776960512 -0700
@@ -39,7 +39,7 @@
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{             0,             0 },		\
 	{      INR_OPEN,     INR_OPEN  },		\
-	{         32768,         32768 },		\
+	{   MLOCK_LIMIT,   MLOCK_LIMIT },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ MAX_SIGPENDING, MAX_SIGPENDING },		\
diff -puN include/asm-x86_64/resource.h~increase-mlock-limit-to-32k-cleanup include/asm-x86_64/resource.h
--- 25/include/asm-x86_64/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.769961576 -0700
+++ 25-akpm/include/asm-x86_64/resource.h	2004-08-05 11:16:01.774960816 -0700
@@ -39,7 +39,7 @@
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{             0,             0 },		\
 	{      INR_OPEN,     INR_OPEN  },		\
-	{         32768,         32768 },		\
+	{   MLOCK_LIMIT,   MLOCK_LIMIT },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ RLIM_INFINITY, RLIM_INFINITY },		\
 	{ MAX_SIGPENDING, MAX_SIGPENDING },		\
diff -puN include/linux/resource.h~increase-mlock-limit-to-32k-cleanup include/linux/resource.h
--- 25/include/linux/resource.h~increase-mlock-limit-to-32k-cleanup	2004-08-05 11:16:01.771961272 -0700
+++ 25-akpm/include/linux/resource.h	2004-08-05 11:16:01.776960512 -0700
@@ -56,6 +56,12 @@ struct rlimit {
 #define _STK_LIM	(8*1024*1024)
 
 /*
+ * GPG wants 32kB of mlocked memory, to make sure pass phrases
+ * and other sensitive information are never written to disk.
+ */
+#define MLOCK_LIMIT	(32*1024)
+
+/*
  * Due to binary compatibility, the actual resource numbers
  * may be different for different linux versions..
  */
_
